daffodil-lib/src/main/scala/org/apache/daffodil/lib/oolag/OOLAG.scala

```diff
@@ -687,22 +697,26 @@ object OOLAG {
 
     final lazy val valueAsAny: Any = {
       if (hasValue) value_.get
-      val res =
-        try {
-          oolagBefore
-          val v = body // good place for a breakpoint
-          oolagAfterValue(v.asInstanceOf[AnyRef])
-          v
-        } catch {
-          case npe: NullPointerException => throw npe
-          case s: scala.util.control.ControlThrowable => throw s
-          case u: UnsuppressableException => throw u
-          case e: Error => throw e
-          case th: Throwable => oolagCatch(th)
-        } finally {
-          oolagFinalize
-        }
-      res
+      else {
```

Comment:
This is the other significant functional improvement. 
The improvement here was minor at best. Not measurable really. Still it is more correct now.